src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php

```diff
@@ -823,6 +824,28 @@ public function testDestruct()
         $this->assertLessThan(4, $duration);
     }
 
+    public function testHandleIsRemovedOnException()
+    {
+        $client = $this->getHttpClient(__FUNCTION__);
+
+        try {
+            $client->request('GET', 'http://localhost:8057/404/not-json');
+            $this->fail(ClientExceptionInterface::class.' expected');
+        } catch (ClientExceptionInterface $e) {
+            // The response content-type mustn't be json as that calls getContent
+            // @see src/Symfony/Component/HttpClient/Exception/HttpExceptionTrait.php:58
+            $this->assertStringNotContainsString('json', $e->getResponse()->getHeaders(false)['content-type'][0]);
+
+            $r = new \ReflectionProperty($client, 'multi');
```

Comment:
This test case should be move to the component: contracts contains only tests that are independent from the implementation.

But I would prefer not testing internal implementation details actually. They don't prevent regressions.

Reply:
I don't have a better idea, so let's keep as is. Just move the test please.

Reply:
Done